src/context.cc

```diff
@@ -263,30 +249,46 @@ void ContextBase::onForeignFunction(uint32_t foreign_function_id, uint32_t data_
 }
 
 FilterStatus ContextBase::onNetworkNewConnection() {
-  CHECK_NET(on_new_connection_, FilterStatus::Continue, FilterStatus::StopIteration);
-  DeferAfterCallActions actions(this);
-  if (wasm_->on_new_connection_(this, id_).u64_ == 0) {
+  CHECK_FAIL2(WasmStreamType::Downstream, WasmStreamType::Upstream, FilterStatus::Continue,
```

Comment:
There is something odd about the fact that this is the only callback where both streams are closed. What's the reason for that? And what's the reason to closing only a single stream in other cases?

Reply:
initially I had some reason, but I cannot recall now.. yeah it seems weird now to me as well, so let me change to close both direction for http too.